c_api/index_io_c.h

```diff
@@ -16,6 +16,7 @@
 #include "Index_c.h"
 #include "VectorTransform_c.h"
 #include "faiss_c.h"
+#include "impl/io_c.h"
```

Comment:
`impl/io_c.h` is not C compatible either, so it cannot be included here.
One would need to either simplify `impl/io_c.h` so that it only uses code that is compatible with both C and C++, or add the minimum declarations necessary inline in the file and remove the `#include`.

Reply:
Made `impl/io_c.h` compatible with C now..